polaris-react/src/utilities/event-listener/hooks.ts

```diff
@@ -0,0 +1,21 @@
+import {useEffect, useRef} from 'react';
+
+export type ElementType = Window | Document | HTMLElement;
```

Comment:
```suggestion
type ElementType = Window | Document | HTMLElement;
```

Thoughts on holding off exporting this for now. I have a local POC enhancing the TypeScript behavior referring to these as `UseEventTarget`s (Matching the [EventTarget.addEventListener docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener))